sql/colexec/colexecbase: TestSimpleProjectOpWithUnorderedSynchronizer failed

sql/colexec/colexecbase.TestSimpleProjectOpWithUnorderedSynchronizer [failed](https://teamcity.cockroachdb.com/viewLog.html?buildId=3508126&tab=buildLog) with [artifacts](https://teamcity.cockroachdb.com/viewLog.html?buildId=3508126&tab=artifacts#/) on master @ [d62a975591ff372daae2f1be56a03109539828b4](https://github.com/cockroachdb/cockroach/commits/d62a975591ff372daae2f1be56a03109539828b4):


```
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:482 +0x1c5
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithFn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:659 +0x379
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithoutAllNullsInjectionWithErrorHandler()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:476 +0x1a7
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithoutAllNullsInjection()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:446 +0x114
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase_test.TestSimpleProjectOpWithUnorderedSynchronizer()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project_test.go:117 +0xba5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1193 +0x202

Goroutine 59 (running) created at:
  github.com/cockroachdb/cockroach/pkg/sql/colexec.(*ParallelUnorderedSynchronizer).init()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/parallel_unordered_synchronizer.go:214 +0x204
  github.com/cockroachdb/cockroach/pkg/sql/colexec.(*ParallelUnorderedSynchronizer).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/parallel_unordered_synchronizer.go:337 +0x647
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*simpleProjectOp).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project.go:125 +0x75
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils.(*vectorTypeEnforcer).Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecutils/operator.go:137 +0x5b
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.constInt64Op.Next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/const.eg.go:391 +0x5e
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase.(*constInt64Op).Next()
      <autogenerated>:1 +0xce
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.(*OpTestOutput).next()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:1214 +0x190
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.(*OpTestOutput).VerifyAnyOrder.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:1265 +0x4e
  github.com/cockroachdb/cockroach/pkg/sql/colexecerror.CatchVectorizedRuntimeError()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexecerror/error.go:91 +0x72
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.(*OpTestOutput).VerifyAnyOrder()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:1264 +0xd7
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithoutAllNullsInjectionWithErrorHandler.func1()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:482 +0x1c5
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithFn()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:659 +0x379
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithoutAllNullsInjectionWithErrorHandler()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:476 +0x1a7
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils.RunTestsWithoutAllNullsInjection()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexectestutils/utils.go:446 +0x114
  github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase_test.TestSimpleProjectOpWithUnorderedSynchronizer()
      /go/src/github.com/cockroachdb/cockroach/pkg/sql/colexec/colexecbase/simple_project_test.go:117 +0xba5
  testing.tRunner()
      /usr/local/go/src/testing/testing.go:1193 +0x202
==================
    simple_project_test.go:129: -- test log scope end --
test logs left over in: /go/src/github.com/cockroachdb/cockroach/artifacts/logTestSimpleProjectOpWithUnorderedSynchronizer969684703
    testing.go:1092: race detected during execution of test
--- FAIL: TestSimpleProjectOpWithUnorderedSynchronizer (0.10s)
```
<details><summary>Reproduce</summary>
<p>
To reproduce, try:

```bash
make stressrace TESTS=TestSimpleProjectOpWithUnorderedSynchronizer PKG=./pkg/sql/colexec/colexecbase TESTTIMEOUT=5m STRESSFLAGS='-timeout 5m' 2>&1
```
Parameters in this failure:

- GOFLAGS=-race -parallel=4
</p>
</details>
/cc @cockroachdb/sql-queries yuzefovich
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestSimpleProjectOpWithUnorderedSynchronizer.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/colexec/colexectestutils/utils.go
+++ b/pkg/sql/colexec/colexectestutils/utils.go
@@ -641,11 +641,11 @@ func RunTestsWithFn(
 			inputSources := make([]colexecop.Operator, len(tups))
 			var inputTypes []*types.T
 			if useSel {
-				rng, _ := randutil.NewTestRandFromGlobalSeed()
 				for i, tup := range tups {
 					if typs != nil {
 						inputTypes = typs[i]
 					}
+					rng, _ := randutil.NewTestRandFromGlobalSeed()
 					inputSources[i] = newOpTestSelInput(allocator, rng, batchSize, tup, inputTypes)
 				}
 			} else {

